chirp/drivers/radtel_rt620.py

```diff
@@ -0,0 +1,545 @@
+            val = 0
+        else:
+            val = _mem.vfo.a.step
+        rs = RadioSetting("vfo.a.step", "Tuning Step",
+                          RadioSettingValueList(baofeng_uv17Pro.LIST_STEPS,
+                                                current_index=val))
+        vfoA.append(rs)
+        if _mem.vfo.b.step >= len(self.STEPS):
+            val = 0
+        else:
+            val = _mem.vfo.b.step
+        rs = RadioSetting("vfo.b.step", "Tuning Step",
+                          RadioSettingValueList(baofeng_uv17Pro.LIST_STEPS,
+                                                current_index=val))
+        vfoB.append(rs)
+        rs = RadioSetting("vfo.a.bcl", "Busy Channel Lock",
+                          RadioSettingValueBoolean(_mem.vfo.a.bcl))
+        vfoA.append(rs)
+        rs = RadioSetting("vfo.b.bcl", "Busy Channel Lock",
+                          RadioSettingValueBoolean(_mem.vfo.b.bcl))
+        vfoB.append(rs)
+        rs = RadioSetting("vfo.a.compander", "Compander",
+                          RadioSettingValueBoolean(_mem.vfo.a.compander))
+        vfoA.append(rs)
+        rs = RadioSetting("vfo.b.compander", "Compander",
+                          RadioSettingValueBoolean(_mem.vfo.b.compander))
+        vfoB.append(rs)
+        workmode.append(vfoA)
+        workmode.append(vfoB)
+
+    def _get_raw_memory(self, number):
+        if number == -1:
+            return self._memobj.vfo.b
+        elif number == -2:
+            return self._memobj.vfo.a
+        else:
+            return super()._get_raw_memory(number)
+
+    def get_memory(self, number):
+        #        if isinstance(number, str):
+        #            number = self._get_specials()[number]
+        mem = super().get_memory(number)
+#          mem.number = name
+        _mem = self._get_raw_memory(number)
+#        if _mem.rxtone == 0:
+#            _mem.rxtone = 0xFFFF
+#        if _mem.txtone == 0:
+#            _mem.txtone = 0xFFFF
+        if hasattr(mem.extra, "keys") and \
+                "compander" not in mem.extra.keys():
+            rs = RadioSetting("compander", "Compander",
+                              RadioSettingValueBoolean(_mem.compander))
+            mem.extra.append(rs)
+        mem.tuning_step = self.STEPS[_mem.step] or self.STEPS[0]
+        if _mem.wide < len(self.MODES):
+            mem.mode = self.MODES[_mem.wide]
+#       if isinstance(mem.number, str):
+#           mem.vfo=1
+        return mem
+
+    def set_memory(self, mem):
+        _mem = self._get_raw_memory(mem.number)
+        super().set_memory(mem)
+        _mem.step = self.STEPS.index(mem.tuning_step)
+        _mem.wide = self.MODES.index(mem.mode)
+        if _mem.rxtone == 0:
+            _mem.rxtone = 0xFFFF
+        if _mem.txtone == 0:
+            _mem.txtone = 0xFFFF
+
+    def get_features(self):
+        rf = super().get_features()
+        rf.has_tuning_step = True
+        rf.valid_tuning_steps = self.STEPS
```

Comment:
Can you just convert the base class to do this the same way it does `valid_modes` below? Then you can just define `STEPS` on your class (as you have done here) and be good.

Reply:
I originally did not want to touch the other source files, so that I do not break something somewhere in inheritance (as this was a standalone test driver not touching anything else) but it is probably a good idea. Only I am not sure where else it is inherited, will have to check.